examples/tutorials/nb_python/managed_rigid_object_tutorial.py

```diff
@@ -0,0 +1,560 @@
+    args, _ = parser.parse_known_args()
+    show_video = args.show_video
+    make_video = args.make_video
+    if make_video and not os.path.exists(output_path):
+        os.mkdir(output_path)
+
+    # %%
+    # [initialize]
+    # create the simulators AND resets the simulator
+
+    cfg = make_configuration()
+    try:  # Got to make initialization idiot proof
+        sim.close()
+    except NameError:
+        pass
+    sim = habitat_sim.Simulator(cfg)
+    agent_transform = place_agent(sim)
+
+    # get the primitive assets attributes manager
+    prim_templates_mgr = sim.get_asset_template_manager()
+
+    # get the physics object attributes manager
+    obj_templates_mgr = sim.get_object_template_manager()
+
+    # get the rigid object manager
+    rigid_obj_mgr = sim.get_rigid_object_manager()
+    # [/initialize]
+
+    # %%
+    # [basics]
+
+    # load some object templates from configuration files
+    sphere_template_id = obj_templates_mgr.load_configs(
+        str(os.path.join(data_path, "test_assets/objects/sphere"))
+    )[0]
+
+    # add a sphere to the scene, returns the object
+    sphere_obj = rigid_obj_mgr.add_object_by_id(sphere_template_id)
+    # move sphere
+    sphere_obj.translation = np.array([2.50, 0, 0.2])
+
+    # simulate
+    observations = simulate(sim, dt=1.5, get_frames=make_video)
+
+    if make_video:
+        vut.make_video(
+            observations,
+            "rgba_camera_1stperson",
+            "color",
+            output_path + "sim_basics",
+            open_vid=show_video,
+        )
+
+    # [/basics]
+    rigid_obj_mgr.remove_all_objects()
+    # %%
+    # [dynamic_control]
+
+    observations = []
+    obj_templates_mgr.load_configs(str(os.path.join(data_path, "objects")))
+    # search for an object template by key sub-string
+    cheezit_template_handle = obj_templates_mgr.get_template_handles(
+        "data/objects/cheezit"
+    )[0]
+    # build multiple object initial positions
+    box_positions = [
+        np.array([2.39, -0.37, 0]),
+        np.array([2.39, -0.64, 0]),
+        np.array([2.39, -0.91, 0]),
+        np.array([2.39, -0.64, -0.22]),
+        np.array([2.39, -0.64, 0.22]),
+    ]
+    box_orientation = mn.Quaternion.rotation(
+        mn.Rad(math.pi / 2.0), np.array([-1.0, 0, 0])
+    )
+    # instance and place the boxes
+    boxes = []
+    for b in range(len(box_positions)):
+        boxes.append(rigid_obj_mgr.add_object_by_handle(cheezit_template_handle))
+        boxes[b].translation = box_positions[b]
+        boxes[b].rotation = box_orientation
+
+    # get the object's initialization attributes (all boxes initialized with same mass)
+    object_init_template = boxes[0].creation_attributes
+    # anti-gravity force f=m(-g)
+    anti_grav_force = -1.0 * sim.get_gravity() * object_init_template.mass
+
+    # throw a sphere at the boxes from the agent position
+    sphere_template = obj_templates_mgr.get_template_by_ID(sphere_template_id)
```

Comment:
Looks like sometimes `ID` is capitalized and other times it isn't.  We should be consistent, my vote is uncapitalized to better fit with python style.

Reply:
@erikwijmans id is a python builtin though, so when naming args as id it will cause issues. :(

Reply:
I don't like the overloading of the term ID anyhow, key or something would be better in the longrun, no? or alternatively uuid to be consistent with Sensors etc?

Reply:
`index`? :)

Reply:
Yeah, for arg names and properties `object_id` makes sense, but I don't think we need to be that explicit in method names.  I don't have a strong preference between id and uuid.  uuid wouldn't be confused with python's builtin `id()` (but no one uses that in our world AFAIK) but no one uses uuid in our world either -- e.g. we reference to things by their instance id in instance segmentation, not their instance uuid -- so uuid isn't a natural choice.

Reply:
Agreed we should pick one consistent naming scheme. In this case the id's refer to unique integer values rather than string keys/handles, so we should keep that distinction. Overlap with python id() is unfortunate, but semantically correct. uuid, object_id, hab_id, etc... would be my preference.